lib/auto-languageclient.ts

```diff
@@ -353,22 +380,19 @@ export default class AutoLanguageClient {
     return newServer;
   }
 
-  private captureServerErrors(childProcess: LanguageServerProcess, projectPath: string): void {
-    childProcess.on('error', (err) => this.handleSpawnFailure(err));
-    childProcess.on('exit', (code, signal) => this.logger.debug(`exit: code ${code} signal ${signal}`));
-    childProcess.stderr.setEncoding('utf8');
-    childProcess.stderr.on('data', (chunk: Buffer) => {
-      const errorString = chunk.toString();
-      this.handleServerStderr(errorString, projectPath);
-      // Keep the last 5 lines for packages to use in messages
-      this.processStdErr = (this.processStdErr + errorString)
-        .split('\n')
-        .slice(-5)
-        .join('\n');
-    });
+  private captureServerErrors(lsProcess: LanguageServerProcess, projectPath: string): void {
+    lsProcess.on('error', (err) => this.onSpawnError(err));
+    lsProcess.on("close", (code, signal) => this.onSpawnClose(code, signal));
+    lsProcess.on("disconnect", () => this.onSpawnDisconnect());
+    lsProcess.on('exit', (code, signal) => this.onSpawnExit(code, signal));
```

Comment:
A little nit picky but mixing double and single quotes. Maybe after this is merged we should add prettier or a linting rule.

Reply:
Ah right. It is hard to maintain the style without Prettier. I will make a new PR after this fixing the format.